### PR TITLE
Fix/persistent queue

### PR DESCRIPTION
**Pull request by wojtekzyla**
Things included in this PR:

- enable persistent queues configuration in the helm chart
- while creating ACLs in the initContainer in the daemonset, change the default user and group of the otel image to 999
- add post delete hook which deletes ACLs set for directories used by agent, gateway and clusterReceiver while those pods run as non-root users

### CODE REVIEW

File: helm-charts/splunk-otel-collector/values.yaml [hunk 1]
```diff
@@ -99,6 +99,18 @@ splunkPlatform:
     #   requests_per_second is the average number of requests per seconds.
     queueSize: 5000
 
+  # When enabled, it uses file_storage extension to persist the queue data.
+  # This can be used to prevent data-loss during restart
+  # NOTE: The File Storage extension will persist state to the node's local file system.
+  # While using the persistent queue it is advised to increase memory limit for agent (agent.resources.limits.memory)
+  # to 1Gi.
+  persistentQueueEnabled:
```
Comment (rmfitzpatrick) [inline helm-charts/splunk-otel-collector/values.yaml:107]: Does it make more sense to nest this in `sendingQueue` since that's what uses it?

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 2]
```diff
@@ -11,6 +11,11 @@ extensions:
     directory: {{ .Values.logsCollection.checkpointPath }}
   {{- end }}
 
+  {{- if or .Values.splunkPlatform.persistentQueueEnabled.logs .Values.splunkPlatform.persistentQueueEnabled.metrics }}
```
Comment (rmfitzpatrick) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:14]: you defined the `"splunk-otel-collector.persistentQueueEnabled"` template for this purpose

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 3]
```diff
@@ -614,6 +619,9 @@ service:
     {{- if and (eq (include "splunk-otel-collector.logsEnabled" .) "true") (eq .Values.logsEngine "otel") }}
     - file_storage
     {{- end }}
+    {{- if or .Values.splunkPlatform.persistentQueueEnabled.logs .Values.splunkPlatform.persistentQueueEnabled.metrics }}
```
Comment (rmfitzpatrick) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:622]: same template note

File: helm-charts/splunk-otel-collector/templates/daemonset.yaml [hunk 4]
```diff
@@ -135,27 +135,39 @@ spec:
           imagePullPolicy: {{ .Values.image.initPatchLogDirs.pullPolicy }}
           command: ['sh', '-c', '
           mkdir -p {{ .Values.logsCollection.checkpointPath }};
-          chown -Rv {{ $agent.securityContext.runAsUser | default 20000 }}:{{ $agent.securityContext.runAsGroup | default 20000 }} {{ .Values.logsCollection.checkpointPath }};
+          chown -Rv {{ $agent.securityContext.runAsUser | default 999 }}:{{ $agent.securityContext.runAsGroup | default 999 }} {{ .Values.logsCollection.checkpointPath }};
```
Comment (rmfitzpatrick) [inline helm-charts/splunk-otel-collector/templates/daemonset.yaml:138]: I think this is a good change but could be problematic if users have adopted the 20000 id. Can you please add a breaking change note in the changelog?

File: helm-charts/splunk-otel-collector/templates/daemonset.yaml [hunk 5]
```diff
@@ -135,27 +135,39 @@ spec:
...
           chmod -v g+rwxs {{ .Values.logsCollection.checkpointPath }};
           {{ if .Values.logsCollection.containers.enabled -}}
           if [ -d "/var/lib/docker/containers" ];
           then
-              setfacl -n -Rm d:m::rx,m::rx,d:g:{{ $agent.securityContext.runAsGroup | default 20000 }}:rx,g:{{ $agent.securityContext.runAsGroup | default 20000 }}:rx /var/lib/docker/containers;
+              setfacl -n -Rm d:m::rx,m::rx,d:g:{{ $agent.securityContext.runAsGroup | default 999 }}:rx,g:{{ $agent.securityContext.runAsGroup | default 999 }}:rx /var/lib/docker/containers;
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/daemonset.yaml:143]: @wojtekzyla can you please make a separate PR for these changes? They seem unrelated and should be captured in a changelog separately.

**Comment by dmitryax** [comment]
Please rebase
